core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/NoneIndexProperties.java

```diff
@@ -29,6 +29,11 @@ public <Type_> Type_ getProperty(int index) {
         throw new IllegalArgumentException("Impossible state: none index property requested");
     }
 
+    @Override
+    public int maxLength() {
+        throw new IllegalArgumentException("Impossible state: none index property requested");
```

Comment:
Didn't we use rather `IllegalStateException` for this purpose? Especially in this case, as the method has no parameters.